packages/server/tests/acceptance/estimateGasPrecompile.spec.ts

```diff
@@ -0,0 +1,896 @@
+      prefix + EstimatePrecompileContractAddress,
+      EstimatePrecompileContractJson.abi,
+      accounts[2].wallet,
+    );
+
+    estimateContractSigner3 = new ethers.Contract(
+      prefix + EstimatePrecompileContractAddress,
+      EstimatePrecompileContractJson.abi,
+      accounts[3].wallet,
+    );
+
+    estimateContractSigner4 = new ethers.Contract(
+      prefix + EstimatePrecompileContractAddress,
+      EstimatePrecompileContractJson.abi,
+      accounts[4].wallet,
+    );
+
+    //Associating accounts[2] and accounts[3] with fungible token and NFT
+    const tx2_ft = await estimateContractSigner2.associateTokenExternal(
+      accounts[2].wallet.address,
+      tokenAddress,
+      Constants.GAS.LIMIT_1_000_000,
+    );
+    await tx2_ft.wait();
+
+    const tx2_nft = await estimateContractSigner2.associateTokenExternal(
+      accounts[2].wallet.address,
+      nftAddress,
+      Constants.GAS.LIMIT_1_000_000,
+    );
+    await tx2_nft.wait();
+
+    const tx3_ft = await estimateContractSigner3.associateTokenExternal(
+      accounts[3].wallet.address,
+      tokenAddress,
+      Constants.GAS.LIMIT_1_000_000,
+    );
+    await tx3_ft.wait();
+
+    const tx3_nft = await estimateContractSigner3.associateTokenExternal(
+      accounts[3].wallet.address,
+      nftAddress,
+      Constants.GAS.LIMIT_1_000_000,
+    );
+    await tx3_nft.wait();
+
+    //NFT
+    await contract.grantTokenKycExternal(nftAddress, accounts[0].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(nftAddress, accounts[1].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(nftAddress, accounts[2].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(nftAddress, accounts[3].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(nftAddress, accounts[4].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(nftAddress, prefix + EstimatePrecompileContractAddress, { gasLimit: 500_000 });
+
+    //Fungible
+    await contract.grantTokenKycExternal(tokenAddress, accounts[0].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(tokenAddress, accounts[1].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(tokenAddress, accounts[2].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(tokenAddress, accounts[3].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(tokenAddress, accounts[4].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(tokenAddress, prefix + EstimatePrecompileContractAddress, {
+      gasLimit: 500_000,
+    });
+  });
+
+  const baseGasCheck = (estimatedGasValue, expectedValue: number) => {
+    // handle deviation of 20%
+    expect(Number(estimatedGasValue)).to.be.lessThan(expectedValue * 1.4);
+  };
+
+  const isWithinDeviation = (
```

Comment:
@ebadiere, is this similar to what you are working on for checking gas within range?

@nickeynikolovv The idea that I believe @ebadiere is working on, is to create a shared Util to do something similar across many other tests that struggle due to fluctuations on Gas used.